### PR TITLE
Add configurable keybinds

### PR DESCRIPTION
**Pull request by ByMartrixx**
Un-hardcodes every used key, adds menu to configure key binds to any key, located in `File > Configure Keybinds...`

image
image
image

### CODE REVIEW

**Review by OroArmor — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

**Comment by ByMartrixx** [comment]
Added support for changing the modifier keys used in each keybind
image
image
image

**Review by OroArmor — APPROVED** [review]

File: enigma-swing/src/main/java/cuchaz/enigma/gui/config/KeyBindsConfig.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    private static ConfigSection runningCfg;
+
+    static {
+        KeyBindsConfig.snapshotConfig();
+    }
+
+    // Save the current configuration state so consistent keybinds can be provided
+    // for keybinds that are not configurable during runtime.
+    public static void snapshotConfig() {
```
Comment (2xsaiko) [inline enigma-swing/src/main/java/cuchaz/enigma/gui/config/KeyBindsConfig.java:23]: Hmmm, is it not possible to change all these in the UI immediately when set through the options? Need to take a closer look later

Comment (ByMartrixx) [reply enigma-swing/src/main/java/cuchaz/enigma/gui/config/KeyBindsConfig.java]: I managed to make it work without issues

**Review by 2xsaiko — APPROVED** [review]
LGTM! Very nice.

**Pull request merged**